### ISSUE TITLE
Leverage VAM-kr to determine names and dependencies

### ISSUE THREAD

**Issue by ipkiss42** [Shougo/neobundle.vim#117]
Hi,

The VAM plugin (MarcWeber/vim-addon-manager) works uses VAM-kr (MarcWeber/vim-addon-manager-known-repositories) as a database of plugins, providing nice names, sources and dependencies among other things. I think it would be very useful to use VAM-kr in NeoBundle as well. Here is how it could work.

Using FuzzyFinder as an example, the current way to install it is:

```
NeoBundle 'bb:ns9tks/vim-fuzzyfinder', {'depends' : 'bb:ns9tks/vim-l9'}
```

This is tedious, because it implies to find the right plugin source (and not one of the forks), to identify the dependencies, and to find again the right plugin source for the dependencies. And every user of FuzzyFinder has to do the same work.

Relying on VAM-kr, it could be as easy as:

```
NeoBundleVamKr 'FuzzyFinder'
```

or maybe:

```
NeoBundle 'FuzzyFinder', {'vam-kr' : 1}
```

To do that, NeoBundle would simply lookup the provided name in VAM-kr, and automatically deduce the source and dependencies (doing the same recursively for each dependencies).
Note that it is still compatible with the 'name' option: you could give another name to use with NeoBundle, for example if you don't like the one in VAM-kr. Same for 'depends' and 'build', if the VAM-kr database is incomplete.

This would be particularly interesting for vim.org plugins, which are quite annoying to install with NeoBundle at the moment.

VAM-kr would be an optional dependency of NeoBundle (just like the dependency on vimproc). If not present, NeoBundle would work as normal, and the NeoBundleVamKr command would return an error.

Additional bonus: it would make it much easier for users currently using VAM to try NeoBundle, because the migration could be (almost) a simple search&replace.

So to summarize the advantages:
1. Users deal with friendly names
2. Users do not need to care about plugin sources
3. Dealing with vim.org scripts is not painful anymore
4. Users do not need to care about (non-optional) dependencies in most cases
5. Users can still use NeoBundle options like 'name', 'depends', 'build', etc.
6. Any improvement in VAM-kr automatically benefits to NeoBundle as well
7. The dependency on VAM-kr is optional
8. VAM users can migrate to NeoBundle more easily
